libs/components/editor-core/src/editor/editor.ts

```diff
@@ -479,7 +479,22 @@ export class Editor implements Virgo {
             return false;
         });
     }
-
+    public copyBlock = async (blockId: string, parentId: string) => {
+        let block = await this.getBlockById(blockId);
+        let newBlcok: AsyncBlock;
+        newBlcok = await this.createBlock(block.type, parentId);
+        let parentProperties = await block.getProperties();
+        newBlcok.setProperties(parentProperties);
+        let children = await block.children();
+        if (children.length) {
+            await Promise.all(
+                children.map(async child => {
```

Comment:
The `map` method has semantics that returns something

```suggestion
                children.forEach(async child => {
```